docs/getting_started/using_modin/using_modin_cluster.rst

```diff
@@ -1,132 +1,136 @@
+Executing in a cluster environment
+----------------------------------
 
 .. note::
-   Be careful when using the Ray client to connect to a remote cluster.
-   This connection mode may not work. Known bugs:
+   Be careful when using the `Ray client`_ to connect to a remote cluster.
+   We don't recommend this connection mode, beacuse it may not work. Known bugs:
    - https://github.com/ray-project/ray/issues/38713,
    - https://github.com/modin-project/modin/issues/6641.
 
-Using Modin on a Ray Cluster
-----------------------------
-
-Now that we have a Ray cluster up and running, we can use Modin to perform pandas
-operation as if we were working with pandas on a single machine. We test Modin's
-performance on the 200MB `NYC Taxi dataset`_ that was provided as part of our
-`Modin's cluster setup config`_. We can time the following operation in a Jupyter
-notebook:
+Modin lets you instantly speed up your workflows with a large data by scaling pandas
+on a cluster. In this tutorial, we will use a 12.5 GB ``big_yellow.csv`` file that was
+created by concatenating a 200MB `NYC Taxi dataset`_ file 64 times. Preparing this
+file was provided as part of our `Modin's Ray cluster setup config`_.
 
-.. code-block:: python
+If you want to use the other dataset, you should provide it to each of
+the cluster nodes with the same path. We recomnend doing this by customizing the
+``setup_commands`` section of the `Modin's Ray cluster setup config`_.
 
-   %%time
-   df = pd.read_csv("big_yellow.csv", quoting=3)
+To run any script in a remote cluster, you need to submit it to the Ray. In this way,
+the script file is sent to the the remote cluster head node and executed there. 
 
-   %%time
-   count_result = df.count()
+In this tutorial, we provide the `exercise_5.py`_ script, which reads the data from the
+CSV file and executes such pandas operations as count, groupby and map.
+As a result of the script, you will see the size of the file being read and the execution
+time of each function.
 
-   %%time
-   groupby_result = df.groupby("passenger_count").count()
+.. note::
+   Some Dataframe functions are executed asynchronously, so to correctly measure execution time 
+   of each function we need to wait for the execution result. We use the special ``execute`` function for this, 
+   but you shouldn't use it in a real case scenario.
 
-   %%time
-   apply_result = df.map(str)
+You can submit this script to the existing remote cluster by running the following command.
 
-.. note::
-   When using local paths, make sure that they are available on all nodes in the
-   cluster, for example using distributed file system like NFS.
+.. code-block:: bash
 
-Modin performance scales as the number of nodes and cores increases. The following
-chart shows the performance of the above operations with 2, 4, and 8 nodes, with
-improvements in performance as we increase the number of resources Modin can use.
+   ray submit modin-cluster.yaml exercise_5.py
```

Comment:
Will the script be executed upon submit right away or we have to manually run it in the cluster?

Reply:
We don't need to do anything other run this command.